app/controllers/clients_controller.rb

```diff
@@ -0,0 +1,46 @@
+class ClientsController < ApplicationController
+  def index
+    @client = Client.all
+  end
+  
+  def new
+    @client = Client.new
+  end
+  
+  def create
+    @client = Client.new(client_params)
+      if @client.save
+        flash[:success] = "Client created successfully!"
+        redirect_to clients_path
+      else
+        render 'new'
+      end
+  end
+  
+  def show
+    @client = Client.find(params[:id])
+  end
+  
+  def destroy
+    @client = Client.find(params[:id])
```

Comment:
DRY

Reply:
Have created a before_action for this part of the code.